WASM.md

```diff
@@ -1,23 +1,64 @@
-Proof of Concept WebAssembly Support.
+WebAssembly Extension Support
 
-# WebAssembly VM
+# Dependencies
 
-The WebAssembly VM is https://github.com/WAVM/WAVM. This VM was chosen for the
-proof of concept because:
+## apt-get install libglib2.0-dev (on Debian/Ubuntu).
 
-  - It is OSS with a permissive license.
-  - It compiles with LLVM for high performance.
-  - It can attach pre-compiled code to the WASM files (e.g. during distribution
-      from a central controller) for faster startup and reduced load on a pool
-      of proxies.
+## emscripten
 
-# TODO
+git clone https://github.com/emscripten-core/emsdk.git
+cd emsdk
+./emsdk install sdk-1.38.25-64bit
+./emsdk activate sdk-1.38.25-64bit
 
-  - Check if the thread is running in the VM in the WAVM Runtime signal handlers and
-    if not call the underlying signal handler from Envoy or add the WAVM signal handler
-    as a pre-check to the Envoy signal handler with the same check.
-  - Add support to WAVM for multple stacks for coroutines/streams with thread shared state.
-    - evaluate async emscripten: https://github.com/kripken/emscripten/wiki/Asyncify
-    - evaulate boost::fiber, GNU Pth, getcontext(), setcontext(), swapcontext().
-  - Add support for shared global state.
-  - Intercept logging messages from WAVM and send to Envoy logs.
+. ./emsdk\_env.sh
+
+It is possible later versions will work, e.g.
+./emsdk install latest
+./emsdk activate latest
+However 1.38.25 is known to work.
+
+## clang-7 or clang-8
+
+export CC=clang
+export CXX=clang++
+
+## tools
+
+sudo apt-get install autoconf automake libtool curl make g++ unzip
```

Comment:
Those are all normal Envoy dependencies, I don't think we need to replicate "how to build Envoy" instructions here.

Reply:
done